docs/source/daml2ts/index.rst

```diff
@@ -0,0 +1,290 @@
+
+TypeScript!
+###########
+
+daml2ts : TypeScript of DAML
+============================
+
+Introduction
+------------
+
+``daml2ts`` generates TypeScript for interacting with DAML via the `json-api <../json-api/index.html>`_ .
+
+``daml2ts`` inputs are compiled DAML modules. ``daml2ts`` outputs are TypeScript source files containing definitions modeling the DAML entities found.
+
+The code ``daml2ts`` generates is in terms of the package `@daml/types <https://github.com/digital-asset/daml/tree/master/language-support/ts/daml-types>`_.
+
+Primitives: @daml/types
+-----------------------
+
+To understand the TypeScript code generated by ``daml2ts``, it is helpful to keep in mind this quick review of the primitives provided by @daml/types.
+
+**Interfaces**:
+
+- ``Serializable<T>``
+- ``Template<T extends object, K = unknown>``
+- ``Choice<T extends object, C, R, K = unknown>``
+
+  ..
+        An object that extends the ``Serializable<T>`` interface has a field ``decoder: () => jtv.Decoder<T>``.
+        An object that extends the ``Template<T, K>`` interface has fields ``templateId: string``, ``keyDecoder: ()=> jtv.Decoder<K>``, ``Archive: Choice<T, {}, {}>`` and extends ``Serializable<T>``.
+        An object that extends the ``Choice<T, C, R, K>`` interface has fields ``template: () => Template<T, K>``, ``argumentDecoder: () => jtv.Decoder<C>``, ``resultDecoder: () => jtv.Decoder<R>`` and ``choiceName: string``.
+
+**Types**:
+
++-------------------+------------------+----------------------------------+
+| TypeScript        | DAML equiv.      | TypeScript definition            |
++===================+==================+==================================+
+| ``Unit``          | ``()``           | ``{}``                           |
++-------------------+------------------+----------------------------------+
+| ``Bool``          | ``Bool``         | ``boolean``                      |
++-------------------+------------------+----------------------------------+
+| ``Int``           | ``Int``          | ``string``                       |
++-------------------+------------------+----------------------------------+
+| ``Decimal``       | ``Decimal``      | ``string``                       |
++-------------------+------------------+----------------------------------+
+| ``Numeric``       | ``Numeric n``    | ``string``                       |
++-------------------+------------------+----------------------------------+
+| ``Text``          | ``Text``         | ``string``                       |
++-------------------+------------------+----------------------------------+
+| ``Time``          | ``Time``         | ``string``                       |
++-------------------+------------------+----------------------------------+
+| ``Party``         | ``Party``        | ``string``                       |
++-------------------+------------------+----------------------------------+
+| ``List<T>``       | ``[t]``          | ``T[]``                          |
```

Comment:
I just realized we're using small `t` in DAML and capital `T` in TypeScript. Let's use the same. Since it is meant to be a metavariable, we can use capital `T` in DAML too. We could also switch to Greek tau if that makes you feel more comfortable. Or alpha, beta.

Reply:
Switching to τ and η